apps/os/backend/utils/trial-channel-setup.ts

```diff
@@ -0,0 +1,363 @@
+    }
+
+    // 3. Send Slack Connect invite
+    logger.info(`Sending Slack Connect invite to ${userEmail}`);
+    const inviteResult2 = await slackAPI.conversations.inviteShared({
+      channel: channelId,
+      emails: [userEmail],
+    });
+
+    if (!inviteResult2.ok) {
+      // Check if it's an invalid email error
+      if (inviteResult2.error === "invalid_email" || inviteResult2.error === "user_not_found") {
+        logger.warn(`Email ${userEmail} not found in Slack, will need alternate email`);
+        return {
+          success: false,
+          error: "invalid_email",
+          message: `The email ${userEmail} is not associated with any Slack account. Please provide your Slack email address.`,
+        };
+      }
+
+      logger.error("Failed to send Slack Connect invite", inviteResult2);
+      // Continue anyway - admin can manually invite later
+    }
+
+    // 4. Create or update routing override
+    // Check if this channel already has an override (from a previous trial attempt)
+    const existingChannelOverride = await db.query.slackChannelEstateOverride.findFirst({
+      where: and(
+        eq(schema.slackChannelEstateOverride.slackChannelId, channelId),
+        eq(schema.slackChannelEstateOverride.slackTeamId, iterateTeamId),
+      ),
+    });
+
+    if (existingChannelOverride) {
+      // Update the existing override to point to the new estate
+      logger.info(
+        `Updating existing routing override: channel ${channelId} (was estate ${existingChannelOverride.estateId}) → estate ${userEstateId}`,
+      );
+      await db
+        .update(schema.slackChannelEstateOverride)
+        .set({
+          estateId: userEstateId,
+          reason: `Trial via Slack Connect for ${userName} (${userEmail})`,
+          metadata: {
+            createdVia: "trial_signup",
+            userEmail,
+            userName,
+            channelName,
+            createdAt: new Date().toISOString(),
+            previousEstateId: existingChannelOverride.estateId,
+            updatedAt: new Date().toISOString(),
+          },
+        })
+        .where(eq(schema.slackChannelEstateOverride.id, existingChannelOverride.id));
+    } else {
+      // Create new override
+      logger.info(`Creating routing override: channel ${channelId} → estate ${userEstateId}`);
+      await db.insert(schema.slackChannelEstateOverride).values({
+        slackChannelId: channelId,
+        slackTeamId: iterateTeamId,
+        estateId: userEstateId,
+        reason: `Trial via Slack Connect for ${userName} (${userEmail})`,
+        metadata: {
+          createdVia: "trial_signup",
+          userEmail,
+          userName,
+          channelName,
+          createdAt: new Date().toISOString(),
+        },
+      });
+    }
+
+    logger.info(
+      `Successfully ${channelWasCreated ? "created" : "reused existing"} trial channel for ${userEmail}: ${channelName} → estate ${userEstateId}`,
+    );
+
+    return {
+      success: true,
```

Comment:
just a comment for later - I think we should get rid of all these `success: true`/`success: false` return types, unless we're going to switch to effect or neverthrow and do them properly.